src/NodaTime/Interval.cs

```diff
@@ -155,6 +155,18 @@ public Instant End
         [Pure]
         public bool Contains(Instant instant) => instant >= start && instant < end;
 
+        /// <summary>
+        /// Deconstruct this value into its components.
+        /// </summary>
+        /// <param name="start">The start of the interval.</param>
+        /// <param name="end">The end of the interval.</param>
+        [Pure]
+        public void Deconstruct(out Instant start, out Instant end)
```

Comment:
If we can only have one overload here, let's have the nullable version which will never throw, rather than the non-nullable one.